llvm/lib/Target/AIE/AIE2AddrSpace.h

```diff
@@ -40,6 +42,71 @@ enum class AddressSpaces {
 
 enum class AIEBanks { A, B, C, D, TileMemory };
 
+static inline uint64_t getDefaultMemoryBank() {
+  std::bitset<32> MemoryBanks;
+  MemoryBanks.set(static_cast<unsigned>(AIEBanks::A))
+      .set(static_cast<unsigned>(AIEBanks::B))
+      .set(static_cast<unsigned>(AIEBanks::C))
+      .set(static_cast<unsigned>(AIEBanks::D));
+  return MemoryBanks.to_ulong();
+}
+
+static inline uint64_t getMemoryBanksFromAddressSpace(unsigned AddrSpace) {
+  std::bitset<32> MemoryBanks;
+
+  switch (static_cast<AddressSpaces>(AddrSpace)) {
+  case AddressSpaces::a:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::A));
+    break;
+  case AddressSpaces::b:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::B));
+    break;
+  case AddressSpaces::c:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::C));
+    break;
+  case AddressSpaces::d:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::D));
+    break;
+  case AddressSpaces::ab:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::A))
+        .set(static_cast<unsigned>(AIEBanks::B));
+    break;
+  case AddressSpaces::ac:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::A))
+        .set(static_cast<unsigned>(AIEBanks::C));
+    break;
+  case AddressSpaces::ad:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::A))
+        .set(static_cast<unsigned>(AIEBanks::D));
+    break;
+  case AddressSpaces::bc:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::B))
+        .set(static_cast<unsigned>(AIEBanks::C));
+    break;
+  case AddressSpaces::bd:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::B))
+        .set(static_cast<unsigned>(AIEBanks::D));
+    break;
+  case AddressSpaces::cd:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::C))
+        .set(static_cast<unsigned>(AIEBanks::D));
+    break;
+  case AddressSpaces::TM:
+    MemoryBanks.set(static_cast<unsigned>(AIEBanks::TileMemory));
+    break;
+  default:
+    MemoryBanks.set();
+    break;
+  }
+
+  return MemoryBanks.to_ulong();
+}
+
+static inline bool addrspacesMayAlias(unsigned AS1, unsigned AS2) {
```

Comment:
This function is target independent. However, we need a duplication for every new target.

Reply:
I will move this to AIEBaseAliasAnalysis.h